src/calliope/preprocess/time.py

```diff
@@ -157,7 +157,7 @@ def cluster(data: xr.Dataset, clustering_file: str | Path, time_format: str):
     """
     clustering_timeseries = pd.read_csv(clustering_file, index_col=0).squeeze()
     clustering_timeseries.index = _datetime_index(
-        clustering_timeseries.index, time_format
+        clustering_timeseries.index + " 00:00:00", time_format
```

Comment:
How is this connected to the dtype coercion fix?

Reply:
Aha, I had wondered where these changes had got to... I'll split them out and into another PR